packages/cozy-client/src/helpers/settings.js

```diff
@@ -0,0 +1,182 @@
+import CozyClient from '../CozyClient'
+import fetchPolicies from '../policies'
+import { Q } from '../queries/dsl'
+
+const defaultFetchPolicy = fetchPolicies.olderThan(60 * 60 * 1000)
+
+/**
+ * Query the cozy-app settings corresponding to the given slug and
+ * extract the value corresponding to the given `key`
+ *
+ * @param {CozyClient} client - Cozy client instance
+ * @param {string} slug - the cozy-app's slug containing the setting (can be 'instance' for global settings)
+ * @param {string} key - The name of the setting to retrieve
+ * @returns {Promise<any>} - The value of the requested setting
+ */
+export const getSetting = async (client, slug, key) => {
+  const query = getQuery(slug)
+
+  const currentSettingsResult = await client.query(
+    query.definition(),
+    query.options
+  )
+
+  const currentSettings = normalizeSettings(currentSettingsResult.data)
+
+  return currentSettings[key]
+}
+
+/**
+ * Save the given value into the corresponding cozy-app setting
+ *
+ * This methods will first query the cozy-app's settings before injecting the new value and then
+ * save the new resulting settings into database
+ *
+ * @param {CozyClient} client - Cozy client instance
+ * @param {string} slug - the cozy-app's slug containing the setting (can be 'instance' for global settings)
+ * @param {string} key - The new value of the setting to save
+ * @param {any | ((oldValue) => any)} valueOrSetter - The new value of the setting to save. It can be the raw value, or a callback that should return a new value
+ * @returns {Promise<any>} - The result of the `client.save()` call
+ */
+export const saveAfterFetchSetting = async (
+  client,
+  slug,
+  key,
+  valueOrSetter
+) => {
+  const query = getQuery(slug)
+
+  const currentSettingsResult = await client.query(
```

Comment:
Wouldn't it be relevant to use a`fetchQueryAndGetFromState` here? This would look more like a useQuery by retrieving the cache if it exists

Reply:
Sorry I made the edit but did not pushed it yet seems some discussion are still in progress

I pushed theme here, but CI will fail due to types error: https://github.com/cozy/cozy-client/compare/0af2607eabfdcc3fe08b06f560166123a320b91d..c44f83a86d7b55513e3bdffbbff1ca9edf62cbb6